src/components/VerticalTabs.jsx

```diff
@@ -1,51 +1,38 @@
 import { useNavigate } from 'react-router-dom'
+import { defaultTabs, roleExtraTabs, tabMap } from './tabsConfig'
 
-export default function VerticalTabs({ active, setActive, tabs, role = "individual" }) {
-    const navigate = useNavigate();
+export default function VerticalTabs({ active, setActive, role = "individual" }) {
+  const navigate = useNavigate();
 
-    const handleLogout = () => {
-        localStorage.removeItem("token");
-        navigate("/", { replace: true });
-    };
+  const handleLogout = () => {
+    localStorage.removeItem("token");
+    navigate("/", { replace: true });
+  };
 
-    // Add your role-based extras here
-    const roleBasedTabs = {
-        owner: ["Manage Users", "Analytics"],
-        leader: ["My Cohort"],
-        individual: []
-    };
+  const allKeys = [
+    ...defaultTabs,
+    ...(roleExtraTabs[role] || []),
+  ];
```

Comment:
Dynamic tab rendering is essential for supporting future scalability. It allows us to thoughtfully expand role-specific features while maintaining a clear baseline of access for all users. This approach enables parallel development across roles without disrupting the core functionality of the PIES Check-In application.